File: RepositoryG_LindaBrante_group3/app/src/main/java/com/example/repositoryg__lindabrante_group3/MainActivity.java
package com.example.repositoryg__lindabrante_group3;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private TextView show_text;
    private EditText enter_text;
    private String text;

    public static final String SHARED_PREFS = "sharedPrefs";
    public static final String TEXT = "text";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button save = findViewById(R.id.saveText);
        show_text = findViewById(R.id.show_text);
        enter_text = findViewById(R.id.text_enter);
        Spinner theme_select = findViewById(R.id.select_theme);

        save.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                show_text.setText(enter_text.getText().toString());
                saveData();
            }
        });

        theme_select.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            public void onItemSelected(AdapterView<?> parent, View view, int pos, long id) {
                Object item = parent.getItemAtPosition(pos);
                if(item.toString().equals("Dark theme")){
                    AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
                }else{
                    AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
                }
            }
            public void onNothingSelected(AdapterView<?> parent) {
            }
        });

        loadData();
        updateViews();
    }

    public void goTo2nd(View v){
        Intent goTo2ndIntent = new Intent(this, MainActivity2.class);
        startActivity(goTo2ndIntent);
    }

    public void saveData(){
        SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();

        editor.putString(TEXT, show_text.getText().toString());
        editor.apply();

        Toast.makeText(this, "Text saved in data store", Toast.LENGTH_SHORT).show();
    }

    public void loadData(){
        SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
        text = sharedPreferences.getString(TEXT, null);
    }

    public void updateViews(){
        show_text.setText(text);
    }
}

File: RepositoryG_Janis_petrovs_group3/app/src/main/java/com/example/repositoryg_group3/MainActivity.kt
package com.example.repositoryg_group3

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import androidx.appcompat.app.AppCompatDelegate
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import com.example.repositoryg_group3.DataStoreManager

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val editText = findViewById<EditText>(R.id.editText)
        val saveButton = findViewById<Button>(R.id.saveButton)
        val goToSecondButton = findViewById<Button>(R.id.goToSecondButton)
        val themeSpinner = findViewById<Spinner>(R.id.themeSpinner)

        // Load saved text from DataStore
        CoroutineScope(Dispatchers.IO).launch {
            val savedText = DataStoreManager.getText(this@MainActivity)
            if (savedText != null) {
                editText.setText(savedText)
            }
        }

        // Setup theme spinner
        val themes = arrayOf("Default", "Dark theme", "Light theme")
        val adapter = ArrayAdapter(this, android.R.layout.simple_spinner_item, themes)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        themeSpinner.adapter = adapter
        themeSpinner.onItemSelectedListener = object : android.widget.AdapterView.OnItemSelectedListener {
            override fun onItemSelected(parent: android.widget.AdapterView<*>?, view: android.view.View?, position: Int, id: Long) {
                when (themes[position]) {
                    "Dark theme" -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
                    "Light theme" -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
                    else -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
                }
                delegate.applyDayNight()
            }
            override fun onNothingSelected(parent: android.widget.AdapterView<*>?) {}
        }

        // Save button functionality
        saveButton.setOnClickListener {
            val text = editText.text.toString()
            CoroutineScope(Dispatchers.IO).launch {
                DataStoreManager.saveText(this@MainActivity, text)
            }
        }

        // Go to second activity
        goToSecondButton.setOnClickListener {
            val intent = Intent(this, SecondActivity::class.java)
            startActivity(intent)
        }
    }
}